### ISSUE TITLE
MoveIt Python Library: Fix parameter handling in servo module API

### ISSUE THREAD

**Issue by peterdavidfagan** [ros-planning/moveit2#1969]
## Background

Currently users have to pass the end effector frame name as a parameter to instantiate a teleop device ([see here](https://github.com/peterdavidfagan/moveit2/blob/e496a9744bdbbd3a367818af52e9b6ae7b4d7db4/moveit_py/moveit/servo_client/teleop.py#L55)). This is inefficient as this parameter is stored in the servo config file ([see here](https://github.com/ros-planning/moveit2/blob/cc635471aadfb9446398ece319ae31c6b72bec86/moveit_ros/moveit_servo/config/panda_simulated_config.yaml#L45)). This task is to update this poor API design such that the `ee_frame_name` is pulled automatically from the servo node parameters. 

### 🤔 What you will need to know.

How to read parameters from a ROS 2 node.

Note: a potential complication is when multiple servo nodes are running at once, which one should you pull parameters from? 

**Good luck with your first issue!**

**Comment by Tejal-19** [comment]
Hey, would like to take this up if no one is on it. Can someone please guide me for the same.

**Comment by peterdavidfagan** [comment]
> Hey, would like to take this up if no one is on it. Can someone please guide me for the same.

Happy to provide guidance @abhijelly seems to have already started working on this. There are some changes required on the open pr https://github.com/ros-planning/moveit2/pull/2012.

**Comment by Tejal-19** [comment]
hey, thankyou for reaching back to me. I couldn't actually see any review comments on the PR. Are you talking about the checks or the to do list structure mentioned at the very start of the PR ? I'm sorry to bother for such an easy one.

**Comment by peterdavidfagan** [comment]
Hey @Tejal-19, 

Thanks for following up on this, you should be able to see it below the following [comment](https://github.com/ros-planning/moveit2/pull/2012#issuecomment-1465165121).

<img width="915" alt="Screenshot 2023-03-15 at 19 16 07" src="https://user-images.githubusercontent.com/42982057/225404861-c5b9d964-3aa8-4a50-9e64-722f49b06aa0.png">

**Comment by ibrahiminfinite** [comment]
@peterdavidfagan 
> Note: a potential complication is when multiple servo nodes are running at once, which one should you pull parameters from?

An idea I had while looking at this
When a user is starting a teleop node ,  I assume they would know which servo node they want to send the commands to, hence they can pass in the namespace of the servo node as a launch argument `servo_namespace`
Currently that is "moveit_servo", but this can be set to something else as well for another servo node
And then rather than passing the `ee_frame_name` as a constructor argument, you can get it using rclpy with parameter name as
`servo_namespace + ".ee_frame_name"` after making sure the servo node has started [here](https://github.com/peterdavidfagan/moveit2/blob/e496a9744bdbbd3a367818af52e9b6ae7b4d7db4/moveit_py/moveit/servo_client/teleop.py#L83)

**Comment by ibrahiminfinite** [comment]
@sjahr 
Can you assign this to me ?
This needs to be addressed in the `moveit_py` updates for Servo refactor.

**Comment by peterdavidfagan** [comment]
Thanks @ibrahiminfinite for picking this up.

**Comment by yaswanth1701** [comment]
Hi @peterdavidfagan, If this issue is still open can I work on it?

**Comment by ibrahiminfinite** [comment]
@yaswanth1701 

It is still open, feel free to give it a try.

**Comment by yaswanth1701** [comment]
Hi everyone, I went through the discussion on the previous PR of this issue and found that basically, we want to load the parameter from the launch file under the namespace of the ```servo_node``` into our teleop node for which one would require to create a service client right? please do let me know if I am in heading in the right direction. thank you

**Comment by sea-bass** [comment]
> Since we no longer have the `ee_frame` parameter in servo, does this issue make sense any longer? In terms of pulling the `ee_frame` name from the servo parameters.

Right, now these values should be automatically extracted from the actual planning group (or subgroup) being used at the time.

There is probably some syncing up on the Python module to work with our changes to Servo (or maybe not...), so I'd say we can close this issue and separately handle bringing this module up to date as needed.
